lib/card_generator.rb

```diff
@@ -0,0 +1,17 @@
+require './card' #to call card class to set classes
+
+filename = "cards.txt"
+class CardGenerator
+
+  def initialize(filename)
+    # file = File.open(filename)
+    # array = File.read(filename).split(",")
+    # cards = Card.new(array)
+
+    IO.readlines(filename).each do |line|
+      values = []
+      values = line.split(",")   #.collect(&:strip)
+      cards = Card.new(values[0],values[1],values[2])
```

Comment:
This looks super close! I think if you change cards in line 14 to an array and do something like:
`cards << Card.new(values[0],values[1],values[2])` this should work!

Pop a pry in between lines 11&12 and verify how the lines are coming in, what they look like split, etc. When I do that, it looks how I'd expect it to and can generate a card!